// @nodots-llc/backgammon-types/dist/websocket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessagePriority = exports.ConnectionStatus = void 0;
exports.isWebSocketMessage = isWebSocketMessage;
exports.isWebSocketResponse = isWebSocketResponse;
exports.isWebSocketError = isWebSocketError;
/**
 * Connection status enumeration
 */
var ConnectionStatus;
(function (ConnectionStatus) {
    ConnectionStatus["DISCONNECTED"] = "disconnected";
    ConnectionStatus["CONNECTING"] = "connecting";
    ConnectionStatus["CONNECTED"] = "connected";
    ConnectionStatus["AUTHENTICATED"] = "authenticated";
    ConnectionStatus["ERROR"] = "error";
    ConnectionStatus["RECONNECTING"] = "reconnecting";
})(ConnectionStatus || (exports.ConnectionStatus = ConnectionStatus = {}));
/**
 * Message priority levels
 */
var MessagePriority;
(function (MessagePriority) {
    MessagePriority["LOW"] = "low";
    MessagePriority["NORMAL"] = "normal";
    MessagePriority["HIGH"] = "high";
    MessagePriority["CRITICAL"] = "critical";
})(MessagePriority || (exports.MessagePriority = MessagePriority = {}));
// ========================================
// Exported Type Guards
// ========================================
/**
 * Type guard for WebSocket messages
 */
function isWebSocketMessage(obj) {
    return (typeof obj === 'object' &&
        obj !== null &&
        typeof obj.type === 'string' &&
        typeof obj.timestamp === 'string');
}
/**
 * Type guard for WebSocket responses
 */
function isWebSocketResponse(obj) {
    return (typeof obj === 'object' &&
        obj !== null &&
        typeof obj.success === 'boolean' &&
        typeof obj.timestamp === 'string');
}
/**
 * Type guard for WebSocket errors
 */
function isWebSocketError(obj) {
    return (typeof obj === 'object' &&
        obj !== null &&
        typeof obj.code === 'string' &&
        typeof obj.message === 'string' &&
        typeof obj.retryable === 'boolean');
}
